### ISSUE TITLE
Focus not restored after returning to library tab

### ISSUE THREAD

**Issue by abaevbog** [zotero/zotero#5683]
https://forums.zotero.org/discussion/128761/zotero-8-beta-focus-does-not-return-to-center-pane-after-closing-tab

1. Navigate to an item in itemTree
2. Press Enter to open it in a new tab
3. Press Cmd+W to close the newly opened tab
4. Note that focus in library tab is no longer on the itemTree but on the "New Collection" button

**Comment by abaevbog** [comment]
Yeah, this is a post notes-in-tabs regression. From what I see, when a tab is selected, context pane tries to focus its content:

https://github.com/zotero/zotero/blob/58baa44ad446fe6dea2ca30a6c104aa28495c987/chrome/content/zotero/elements/contextPane.js#L263-L282

It ends up emitting `tab-focus` even that reader or note tabs know how to handle. For the library tab, though, we should be using the logic of refocusing `ZoteroPane.lastFocusedElement`, with itemTree as the default fallback. That currently lives in `Zotero_Tabs.tabhooks.refocus.library` - but it seems to be only used as `onRefocus` prop in the tabbar.

https://github.com/zotero/zotero/blob/58baa44ad446fe6dea2ca30a6c104aa28495c987/chrome/content/zotero/tabs.js#L148-L165

**Comment by windingwind** [comment]
Thanks for the investigation.

I think the described behavior of focusing the new collection button is from the `Zotero_Tabs.tabHooks.focus`, which is also the logic inherited from the previous code (just unified into one place). The focus code in the context pane is another old deeply coupled code that I tried not to touch during refactoring. Though it actually calls the `setFocus`, which is also what gets called by `tab-focus` event. This is likely harmless.

Since we are seeing various issues with the tab focus and we used to have multiple places for tab focus handling and they seem to work by sometimes setting timeout to overwrite each other, I think it's time to further integrate them into one place.

Not sure if I summarized it correctly:

When you select a tab:

- It's a newly created tab, or the tab is selected with `keepTabFocused !== true`, trigger the behavior in `tabHooks.focus`, e.g. the library's first button
- If it's not library tab and context pane note editor is active, keep focus
- Otherwise, use tabContent's `tab-focus` handle to let the tab decide where to focus
- A special case: when you navigate tabs with the arrow keys, keep focus in tabs.

Let me know if I missed any part.
